src/logging.ts

```diff
@@ -401,6 +401,26 @@ export class Logger {
     return `[${this.name}] ${message}`;
   }
 }
+/** Utility function to return permissions for a given file */
+function getFilePermissions(filePath: string): {
```

Comment:
I went back and forth regarding all these lines of code for what ultimately is readability when you're looking at the templated string, but decided it was worth it.